File: check_tags.py
import re

with open('src/app/components/TeacherDashboard.tsx', 'r', encoding='utf-8') as f:
    text = f.read()

lines = text.split('\n')
div_count = 0
main_count = 0

for i, line in enumerate(lines):
    opens = len(re.findall(r'<div[\s>]', line))
    closes = len(re.findall(r'</div\s*>', line))
    div_count += opens
    div_count -= closes
    
    m_opens = len(re.findall(r'<main[\s>]', line))
    m_closes = len(re.findall(r'</main\s*>', line))
    main_count += m_opens
    main_count -= m_closes
    
    if main_count > 0 and div_count < 0:
        print(f"Negative div count at {i+1} : {line}")

print('Final Div:', div_count)
print('Final Main:', main_count)

File: check_tags.js
import fs from 'fs';
const text = fs.readFileSync('src/app/components/TeacherDashboard.tsx', 'utf8');

const lines = text.split('\n');
let divCount = 0;
let mainCount = 0;

for (let i = 0; i < lines.length; i++) {
  const line = lines[i];
  
  if (line.trim().startsWith('//') || line.trim().startsWith('{/*')) continue;
  
  const opens = (line.match(/<div[\s>]/g) || []).length;
  const selfCloses = (line.match(/<div[^>]*\/>/g) || []).length;
  const closes = (line.match(/<\/div\s*>/g) || []).length;
  
  divCount += opens;
  divCount -= selfCloses;
  divCount -= closes;
  
  const mOpens = (line.match(/<main[\s>]/g) || []).length;
  const mCloses = (line.match(/<\/main\s*>/g) || []).length;
  mainCount += mOpens;
  mainCount -= mCloses;
}

console.log('Final Div:', divCount);
console.log('Final Main:', mainCount);
